docs/DeveloperGuide.md

```diff
@@ -1,11 +1,267 @@
 # Developer Guide
+**By:** Team T14-4<br/>
+**Since:** May 2020
 
-## Design & Implementation
+* [1. Design & Implementation](#design-implementation)
+    + [1.1. [Proposed] Search stock feature](#search-stock)
+    + [1.2. [Proposed] List stock in descending quantities](#list-stock)
+    + [1.3. [Proposed] Generate profit for the day](#generate-profit)
+    + [1.4. [Proposed] Search reservation](#search-reservation)
+    + [1.5. [Proposed] Search dish](#search-dish)
+* [2. Product Scope](#product-scope)
+* [3. User Stories](#user-stories)
+* [4. Non-Functional Requirements](#nonfunctional-requirement)
+* [5. Glossary](#glossary)
+
+
+<a name="design-implementation"></a>
+
+## 1. Design & Implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+<a name="search-stock"></a>
+
+### 1.1 [Proposed] Search stock feature
+#### 1.1.1 Proposed implementation
+
+In the restaurant daily report, users can search against the stock category by supplying a keyword.
+
+Given below is an example usage scenario and how the search mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. An empty `stock` will be initialized.
+
+Step 2. The user executes `add stock; i/tomato; q/10; p/$0.40;` command to add a tomato ingredient into the `stock`. Further, the user may add more ingredients into the current `stock`. Suppose the user executes `add stock; i/potato; q/5; p/$0.40;` and `add stock; i/rice; q/3; p/$0.40;` as well.
+
+Step 3. The user can now search against the current `stock` to see if an ingredient is stored in the `stock`. The user now executes `search stock; k/tomato;`, which will display the following result in the image. 
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/59989652/77921374-47b58b00-72d2-11ea-84bd-fe0c790b1d59.PNG">
+</p>
+
+The following sequence diagram shows how the search operation works:
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/59989652/77916616-e8ed1300-72cb-11ea-9ba0-79e1ccc119ab.png">
+</p>
```

Comment:
![sequenceArrowWrong](https://user-images.githubusercontent.com/42376931/77988878-eb914c00-734f-11ea-9c81-2063ce391f76.JPG)
There seems to be some errors in the sequence diagram. The part where the blue arrow is pointing to is just a minor issue where perhaps you should include the return arrow for the constructor method as well since return arrows have been included for all the other method calls.

As for the part pointed to by the red arrow, there is a missing arrow to show the method that is being invoked. Perhaps this should have been where the execute(stock) should be?

Reply:
Yes you are right in the arrows. I have fixed this in my diagram as well. Thanks for pointing that out! :)